File: uke1/Tre.java
package uke1;

public class Tre {
    Tre forelder = null;

    public void setForelder(Tre t){
        this.forelder = t;
    }

    public static int depth(Tre t){
        if (t.forelder == null){
            return -1;
        }
        return 1 + depth(t.forelder);
    }


    public static void main(String[] args) {
        Tre s = new Tre();
        Tre k = new Tre();
        Tre m = new Tre();
        Tre v = new Tre();
        s.setForelder(k);
        k.setForelder(m);
        m.setForelder(v);


        System.out.println(depth(v));
        System.out.println(depth(m));
        System.out.println(depth(k));
        System.out.println(depth(s));


    }
}

File: uke5/Tre.py
class TreNode:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
        self.forelder = None

    def setForelder(self, nyForelder):
        self.forelder = nyForelder

    def depth(self, v):
        if (self.forelder == None):
            return -1
        return 1 + self.depth(self.forelder)

    def insert(self, data):
        if self.data:
            if data < self.data:
                if self.left is None:
                    self.left = TreNode(data)
                else:
                    self.left.insert(data)
            elif data > self.data:
                if self.right is None:
                    self.right = TreNode(data)
                else:
                    self.right.insert(data)
        else:
            self.data = data

    def printTree(self):
        if self.left:
            self.left.printTree()
        print(self.data)
        if self.right:
            self.right.printTree()


root = TreNode(12)
root.insert(6)
root.insert(14)
root.insert(3)
root.printTree()
